### PR TITLE
xilinx: fix zynqmp build when tsp is enabled

### PR DESCRIPTION
**Pull request by sivadur**
Patches to fix issue https://github.com/ARM-software/tf-issues/issues/589

### CODE REVIEW

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Comment by jeenu-arm** [comment]
jenkins: test this please

File: plat/xilinx/zynqmp/aarch64/zynqmp_common.c [hunk 1]
```diff
@@ -205,12 +205,21 @@ static char *zynqmp_get_silicon_idcode_name(void)
...
+	/*
+	 * For BL32, get the chip id info directly by reading corresponding
+	 * registers instead of making pm call. This has limitation
+	 * that these registers should be configured to have access
+	 * from APU which is default case.
+	 */
+	chipid[0] = mmio_read_32(ZYNQMP_CSU_BASEADDR + ZYNQMP_CSU_IDCODE_OFFSET);
+	chipid[1] = mmio_read_32(EFUSE_BASEADDR + EFUSE_IPDISABLE_OFFSET);
+#else
+	if (pm_get_chipid(chipid))
```
Comment (antonio-nino-diaz-arm) [inline plat/xilinx/zynqmp/aarch64/zynqmp_common.c]: There is an issue in line 220:

```[Required] MISRA C-2012 Rule 14.4 violation: Condition "pm_get_chipid(chipid)" is not essentially boolean.```

Change it to:

```
if (pm_get_chipid(chipid) != PM_RET_SUCCESS)
```

**Comment by sivadur** [comment]
@antonio-nino-diaz-arm 
Thanks for review, Fixed your comment about MISRA and also fixed one more issue with patch1 as well. It looks fine now.

**Pull request merged**